constraints/mesh-level-strict-mtls.yaml

```diff
@@ -0,0 +1,6 @@
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: PeerAuthnMeshStrictMtls
+metadata:
+  name: mesh-level-strict-mtls
+spec:
+  enforcementAction: deny
```

Comment:
Good point, but what's the recommendation in this case where the associated `ConstraintTemplate` does that: `data.inventory.namespace[ns][_].PeerAuthentication[_]`. It's not dedicated on a specific match, is it? What's the recommendation here? Not having a `match` expression in the `Constraint`, could it generate performance issue where this constraint is evaluated with all the resources?

Reply:
The recommendation is for all constraints to have a match field. In this case, I think this constraint should match with `Namespace`, and in the rego, we can access PeerAuthentications through Gatekeeper's cache by `data.inventory.namespace[input.review.object.metadata.name]["security.istio.io/v1beta1"].PeerAuthentication[_]`. Note that this won't be able to block any deployment because we can't create a `PeerAuthentication` without creating a `Namespace` first, but this should show up as a violation during audit.

Reply:
Gotcha, thanks for confirming, do you mean that the `enforcementAction` should be `dryrun` instead of `deny` too?

Reply:
Just committed the `dryrun` and `match` `Namespace`, let me know what do you think.